src/inflators/text.ts

```diff
@@ -0,0 +1,129 @@
+import { addComponent } from "bitecs";
+import { BackSide, DoubleSide, FrontSide } from "three";
+import { Text as TroikaText } from "troika-three-text";
+import { HubsWorld } from "../app";
+import { TextTag } from "../bit-components";
+import { addObject3DComponent } from "../utils/jsx-entity";
+
+export type TextParams = {
+  value: string;
+  anchorX?: "left" | "center" | "right";
+  anchorY?: "top" | "top-baseline" | "top-cap" | "top-ex" | "middle" | "bottom-baseline" | "bottom";
+  clipRect?: [number, number, number, number] | null;
+  color?: string;
+  curveRadius?: number;
+  depthOffset?: number;
+  direction?: "auto" | "ltr" | "trl";
+  fillOpacity?: number;
+  fontUrl?: string | null;
+  fontSize?: number;
+  glyphGeometryDetail?: number;
+  gpuAccelerateSDF?: boolean;
+  letterSpacing?: number;
+  lineHeight?: number | "normal";
+  maxWidth?: number;
+  opacity?: number;
+  outlineBlur?: number | `${number}%`;
+  outlineColor?: string;
+  outlineOffsetX?: number | `${number}%`;
+  outlineOffsetY?: number | `${number}%`;
+  outlineOpacity?: number;
+  outlineWidth?: number | `${number}%`;
+  overflowWrap?: "normal" | "break-word";
+  sdfGlyphSize?: number | null;
+  side?: "front" | "back" | "double";
+  strokeColor?: string;
+  strokeOpacity?: number;
+  strokeWidth?: number | `${number}%`;
+  textAlign?: 'left' | 'right' | 'center' | 'justify';
+  textIndent?: number;
+  whiteSpace?: "normal" | "nowrap";
+};
+
+const THREE_SIDES = {
+  front: FrontSide,
+  back: BackSide,
+  double: DoubleSide
+};
+
+// Defaults values must be set for all the optional properties
+// to safely use foo! operator in the inflator.
+const DEFAULTS: TextParams = {
```

Comment:
What I like about using Partial/Required as discussed below is this would be type checked for us so we couldn't accidentally miss a default value.